### DISCUSSION TITLE
remapped post_from_clipboard does not work via ssh

### DISCUSSION THREAD

**Discussion by PgLoLo** [kovidgoyal/kitty#6660] [Q&A] [answered]
I remapped keystroke `ctrl+v` to `post_from_clipboard` in `.config/kitty/kitty.conf` and it works perfectly locally, but does not work via ssh. Instead after two presses of `ctrl+v` "^V" appears in the terminal. At the same time, `ctrl+shift+v` does work via ssh (which I did not remapped).

Moreover, I remapped `ctrl+c` to `copy_to_clipboard` and it works perfectly both locally and via ssh.

Situation does not change whether I use standard `ssh` from kitty, or `kitty +kitten ssh`

What can be the problem and how can I fix it?

Thank you.

Edit: I have just realized, that problem is in bash. So, by default I use fish locally and in fish everything work perfectly fine. But in bush it does not work even locally.

**Comment by kovidgoyal** [comment]
Does not reproduce for me with

kitty --config=NONE -o 'map ctrl+v paste_from_clipboard'

ssh localhost

press ctrl+v

text from clipboard is pasted as expected.

**Reply by PgLoLo** [reply]
That's very interesting. You are absolutely right, and in fact, when I run kitty with your command `kitty --config=NONE -o 'map ctrl+v paste_from_clipboard'` pasting with ctrl+v **works perfectly fine** both via ssh and in bash.

But. I removed everything from `~/.config/kitty/kitty.conf`, this is literally the whole content of the file:
```
map   ctrl+v        past_from_clipboard
```
and then ctrl+v **does not** work via ssh and in bash.

Very weird. What is the difference between running kitty with `kitty.conf` with only this keymap and running kitty via `kitty --config=NONE -o 'map ctrl+v paste_from_clipboard'`?

**Comment by kovidgoyal** [comment, marked as answer]
There is no difference and you spelled paste_from_clipboard wrong in
your kitty.conf

On Thu, Oct 05, 2023 at 11:56:38AM -0700, PgLoLo wrote:
> That's very interesting. You are absolutely right, and in fact, when I run kitty with your command `kitty --config=NONE -o 'map ctrl+v paste_from_clipboard'` pasting with ctrl+v **works perfectly fine** both via ssh and in bash.
> 
> But. I removed everything from `~/.config/kitty/kitty.conf`, this is literally the whole content of the file:
> ```
> map   ctrl+v        past_from_clipboard
> ```
> and then ctrl+v **does not** work via ssh and in bash.
> 
> Very weird. What is the difference between running kitty with `kitty.conf` with only this rempa and running kitty via `kitty --config=NONE -o 'map ctrl+v paste_from_clipboard'`?
> 
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6660#discussioncomment-7202108
> You are receiving this because you commented.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by PgLoLo** [reply]
Such an obvious mistake. Thank you very much for your help!
